### PR TITLE
New TC for PC Test on higher lag ids

### PR DESCRIPTION
**Pull request by vikshaw-Nokia**
### Description of PR

Summary:
The Test Cases focuses on the below PR Changes of lagid allocation to portchannel and new SYSTEM_LAG_IDS_FREE_LIST 

[[lag_id] Add lagid to free_list when LC absent for 30 minutes by mlok-nokia · Pull Request #542 · sonic-net/sonic-platform-daemons (github.com)](https://github.com/sonic-net/sonic-platform-daemons/pull/542)
[[Chassis][LAG_ID] Address the same lagid been used in two different LCs issue by mlok-nokia · Pull Request #3303 · sonic-net/sonic-swss (github.com)](https://github.com/sonic-net/sonic-swss/pull/3303)
[[database-chassis][lagid] Initialize SYSTEM_LAG_IDS_FREE_LIST in CHASSIS_APP_DB by mlok-nokia · Pull Request #20369 · sonic-net/sonic-buildimage (github.com)](https://github.com/sonic-net/sonic-buildimage/pull/20369)

This PR is dependent on the above mentioned PR.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [x] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
Verification of the new feature of LAG ID allocation to PortChannel and new list of SYSTEM_LAG_IDS_FREE_LIST 

#### How did you do it?
The test case verifies traffic flow between all port channels in a full mesh topology. It begins with a system reboot to allocate the lower series of LAG IDs and validate the SYSTEM_LAG_IDS_FREE_LIST. Next, the LAG IDs are incremented through temporary port channel additions and deletions. A configuration reload is then performed to allocate higher LAG IDs to the port channels, followed by a final validation of mesh traffic. 

#### How did you verify/test it?
Verified on T2 VOQ Chassis

#### Any platform specific information?
Applicable for T2 Chassis

#### Supported testbed topology if it's a new test case?
T2 Chassis

### Documentation

[[lag_id] Add lagid to free_list when LC absent for 30 minutes by mlok-nokia · Pull Request #542 · sonic-net/sonic-platform-daemons (github.com)](https://github.com/sonic-net/sonic-platform-daemons/pull/542)
[[Chassis][LAG_ID] Address the same lagid been used in two different LCs issue by mlok-nokia · Pull Request #3303 · sonic-net/sonic-swss (github.com)](https://github.com/sonic-net/sonic-swss/pull/3303)
[[database-chassis][lagid] Initialize SYSTEM_LAG_IDS_FREE_LIST in CHASSIS_APP_DB by mlok-nokia · Pull Request #20369 · sonic-net/sonic-buildimage (github.com)](https://github.com/sonic-net/sonic-buildimage/pull/20369)

### CODE REVIEW

**Comment by mlok-nokia** [comment]
@arlakshm @judyjoseph  This mgmt PR is for the new LagId allocation testing.  It tests the middle range LagIds and upper range lagIds

File: tests/pc/test_po_update.py [hunk 1]
```diff
@@ -374,3 +376,214 @@ def del_add_members():
...
+
+    # Ensure BGP sessions are re-established after reload
+    pytest_assert(wait_until(300, 10, 0,
+                             duthost.check_bgp_session_state_all_asics, up_bgp_neighbors, "established"))
+
+    # Perform another LAG set sanity check
+    lag_set_sanity(duthosts)
+
+    # Send data after the configuration reload
+    send_data(dut_mg_facts, duthost, ptfadapter)
```
Comment (arlakshm) [inline tests/pc/test_po_update.py:567]: The lag in the minigraph will always use the higher lag-ids?

File: tests/pc/test_po_update.py [hunk 2]
```diff
@@ -5,12 +5,14 @@
 
 import pytest
 import logging
-
+from tests.common.reboot import reboot
```
Comment (arlakshm) [inline tests/pc/test_po_update.py:8]: nit: sort imports

**Review by arlakshm — COMMENTED** [review]
as comments

File: tests/pc/test_po_update.py [hunk 3]
```diff
@@ -374,3 +376,214 @@ def del_add_members():
...
+        ptfadapter,
+        reload_testbed_on_failed, localhost):
+    """
+       Test port channel traffic with higher lag ids.
+       """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+    # Check if the device is a modular chassis and the topology is T2
+    is_chassis = duthost.get_facts().get("modular_chassis")
+    if not (is_chassis and tbinfo['topo']['type'] == 't2'):
```
[the remark this answers is not in the document]
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:512]: @arlakshm 
Will Incorporate this Comment

File: tests/pc/test_po_update.py [hunk 4]
```diff
@@ -374,3 +376,214 @@ def del_add_members():
...
+
+    # Ensure BGP sessions are re-established after the second reload
+    pytest_assert(wait_until(300, 10, 0,
+                             duthost.check_bgp_session_state_all_asics, up_bgp_neighbors, "established"))
+
+    # Perform a final LAG set sanity check
+    lag_set_sanity(duthosts)
+
+    # Send data one more time after the final sanity check
+    send_data(dut_mg_facts, duthost, ptfadapter)
```
[the remark this answers is not in the document]
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:589]: @arlakshm 
We are Deleting the PC as soon as we create it.
Do you still want something in cleanup?
image

File: tests/pc/test_po_update.py [hunk 5]
```diff
@@ -374,3 +376,214 @@ def del_add_members():
...
+        if len(seen) != (end - start + 1):
+            logging.error(
+                "Missing or extra values are found in SYSTEM_LAG_IDS_FREE_LIST:{} or SYSTEM_LAG_ID_SET:{}".format(
+                    LAG_IDS_FREE_LIST, SYSTEM_LAG_ID_SET))
+            return False
+
+        # Check for duplicate values in both the free and assigned LAG ID lists
+        if any(LAG_IDS_FREE_LIST.count(x) > 1 or SYSTEM_LAG_ID_SET.count(
+                x) > 1 or x in LAG_IDS_FREE_LIST and x in SYSTEM_LAG_ID_SET for x in seen):
+            logging.error(
```
[the remark this answers is not in the document]
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:487]: @arlakshm 
We are asserting it below, with wait_untill.
Kindly let me know, if you wan to assert it in a different way
image

File: tests/pc/test_po_update.py [hunk 6]
```diff
@@ -374,3 +376,214 @@ def del_add_members():
...
+    # Retrieve the list of free LAG IDs from the database
+    LAG_IDS_FREE_LIST = voqdb.dump("SYSTEM_LAG_IDS_FREE_LIST")["SYSTEM_LAG_IDS_FREE_LIST"]['value']
+
+    def verify_system_lag_sanity():
+        # Combine the free LAG IDs and assigned LAG IDs into a set to check for uniqueness
+        seen = set(LAG_IDS_FREE_LIST + SYSTEM_LAG_ID_SET)
+
+        # Verify that the number of LAG IDs seen matches the expected range from start to end
+        if len(seen) != (end - start + 1):
+            logging.error(
```
[the remark this answers is not in the document]
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:479]: @arlakshm 
Same as above.

File: tests/pc/test_po_update.py (hunk 1 above)
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:567]: @arlakshm 
There is no relation to minigraph here.
We are trying to increase the lag ids allocation in free_list by creating and deleting temp PC.
Please let me know if there is any other thoughts regarding this

File: tests/pc/test_po_update.py (hunk 2 above)
Comment (vikshaw-Nokia) [reply tests/pc/test_po_update.py:8]: @arlakshm 
Will Incorporate this Comment

**Comment by vikshaw-Nokia** [comment]
@arlakshm 
Hello, 
Greetings of the day.
Have Incorporated the comments as applicable.
Kindly review the above inline comments. 

Regards,
Vikash

**Comment by arlakshm** [comment]
Waiting for the code PRs to be merged

**Comment by bingwang-ms** [comment]
Approved the new test case cherry-pick as it's only for T2

**Comment by vikshaw-Nokia** [comment]
Hello @arlakshm 
Please help with the merge for 202405 https://github.com/sonic-net/sonic-mgmt/pull/16142

**Pull request merged**